### PR TITLE
Translate location conditions for global templates

### PR DESCRIPTION
**Pull request by strategio**
This will translate the conditional ID when it refers to:
- a post or taxonomy term
- a parent post
- posts which are assigned a specific term

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlpb-162

### CODE REVIEW

File: tests/phpunit/tests/test-wpml-elementor-translate-ids.php [hunk 1]
```diff
@@ -52,6 +56,102 @@ public function it_should_translate_template_id() {
...
+		$sub_id           = mt_rand( 1, 10 );
+		$translate_id     = mt_rand( 11, 20 );
+		$post_type        = 'hotel';
+		$parsed_conditions = array( 'sub_name' => $post_type );
+
+		\WP_Mock::userFunction( 'get_post_type',
+            array(
+                'times' => 0,
+            )
+		);
```
Comment (brucepearson) [inline tests/phpunit/tests/test-wpml-elementor-translate-ids.php]: @strategio Please fix formatting of this block.

File: tests/phpunit/tests/test-wpml-elementor-translate-ids.php [hunk 2]
```diff
@@ -52,6 +56,102 @@ public function it_should_translate_template_id() {
...
+		$translate_id     = mt_rand( 11, 20 );
+		$post_type        = 'hotel';
+		$parsed_conditions = array( 'sub_name' => 'child_of' );
+
+		\WP_Mock::userFunction( 'get_post_type',
+            array(
+                'args'   => $sub_id,
+                'return' => $post_type
+            )
+		);
```
Comment (brucepearson) [inline tests/phpunit/tests/test-wpml-elementor-translate-ids.php]: @strategio Same formatting problem here.

**Review by brucepearson — DISMISSED** [review]

**Comment by strategio** [comment]
@OnTheGoSystems/wpml, review please.

**Comment by strategio** [comment]
One more review please @OnTheGoSystems/wpml

**Comment by strategio** [comment]
@OnTheGoSystems/wpml please review.

**Review by andreasciamanna — DISMISSED** [review]

**Comment by strategio** [comment]
@brucepearson, I thought you approved this PR, but you didn't give a formal vote. Any other comment?

File: src/class-wpml-elementor-translate-ids.php [hunk 3]
```diff
@@ -22,6 +23,30 @@ public function translate_theme_location_template_id( $template_id ) {
...
+	 *
+	 * @return int|string
+	 */
+	public function translate_location_condition_sub_id( $sub_id, $parsed_condition ) {
+		$sub_name = isset( $parsed_condition['sub_name'] ) ? $parsed_condition['sub_name'] : null;
+
+		if ( (int) $sub_id > 0 && $sub_name ) {
+			$element_type = $sub_name;
+
+			if ( 'child_of' === $sub_name ) {
```
Comment (brucepearson) [inline src/class-wpml-elementor-translate-ids.php:45]: @strategio It might be a good idea to add a short comment about what `$sub_name` means.

**Review by brucepearson — DISMISSED** [review]
@strategio This looks good. +1

**Pull request merged**